Task/MCU/Floor1_USART_inputdata/Core/Src/main.c

```diff
@@ -201,6 +235,24 @@ int main(void)
 	      HAL_UART_Transmit(&huart2, (uint8_t *)"\n", sizeof("\n"), 1000);
 
 #endif //ifdef R_T_C
+
+#ifdef I_W_D_G
+	      // Get the elapsed time since starting
+	          elapsedTime = HAL_GetTick() - startTime;
+
+	      // Check if the first 30 seconds have elapsed
+	          if (elapsedTime <= IWDG_TIMEOUT)
+	          {
+	            // Send signal to the watchdog
+	        	HAL_UART_Transmit(&huart2, (uint8_t *)"Health is Okay..!\n", sizeof("Health is Okay..!\n"), 1000);
+	        	HAL_Delay(1000);
+	        	watchdogFeed();
+	          }
+	          else{
+	        	  HAL_UART_Transmit(&huart2, (uint8_t *)"Health signal is stopped..!\n", sizeof("Health signal is stopped..!\n"), 1000);
+	        	  HAL_Delay(1000);
```

Comment:
Why this delay requires here ?

Reply:
when we print it using delay, it will help developer to check whether the watchdog is resetting the microcontroller as per the timeout they mentioned. For example, If developer keeps timeout of 30 sec then the text will print 30 times before microcontroller will reset as i have used delay of 1 sec